src/main/java/org/marketplace/controllers/AdvertisementManagementController.java

```diff
@@ -0,0 +1,24 @@
+package org.marketplace.controllers;
+
+
+import org.marketplace.models.Advertisement;
+import org.marketplace.services.AdvertisementManagementService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/advertisements")
+public class AdvertisementManagementController {
+    private final AdvertisementManagementService advertisementManagementService;
+
+    public AdvertisementManagementController(AdvertisementManagementService advertisementManagementService){
+        this.advertisementManagementService = advertisementManagementService;
+    }
+
+    @PostMapping
+    public Advertisement requestAddOffer(@RequestBody Advertisement advertisement){
```

Comment:
i saw Advertisement uses @Size annotation, i think to make it work we are lacking @Valid annotation

like
`    @PostMapping
    public Advertisement requestAddOffer(@RequestBody @Valid Advertisement advertisement){
        return advertisementManagementService.addAdvertisement(advertisement);
    }`